backend/src/main/java/com/github/clarakraus/susume/blog/user/UserService.java

```diff
@@ -0,0 +1,49 @@
+package com.github.clarakraus.susume.blog.user;
+
+import com.github.clarakraus.susume.blog.Blog;
+import com.github.clarakraus.susume.blog.BlogRepo;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final PasswordEncoder passwordEncoder;
+    private final BlogRepo blogRepo;
+
+    @ResponseStatus(HttpStatus.CREATED)
+    public void createAccount(RegisterDetails registerDetails) {
+        if (!Objects.equals(registerDetails.getPassword(), registerDetails.getPasswordRepeat())) {
+            throw new IllegalArgumentException("passwords do not match");
+        }
+        if (registerDetails.getUsername() == null || registerDetails.getUsername().isBlank()) {
+            throw new IllegalArgumentException("username is blank");
+        }
+        if (registerDetails.getPassword() == null || registerDetails.getPassword().isBlank()) {
+            throw new IllegalArgumentException("password is blank");
+        }
+
```

Comment:
statt findAll und anyMatch müsst hier doch eine findByUsername Suche in der DB reichen